base/shell/explorer/trayclock.cpp

```diff
@@ -529,19 +533,19 @@ LRESULT CTrayClockWnd::OnPaint(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bH
         rcClient.top = (rcClient.bottom - CurrentSize.cy) / 2;
         rcClient.bottom = rcClient.top + CurrentSize.cy;
 
-        for (i = 0, line = 0;
-                i < CLOCKWND_FORMAT_COUNT && line < VisibleLines;
-                i++)
+        if (VisibleLines == 2 && !g_TaskbarSettings.bShowWeekday)
```

Comment:
If ```VisibleLines``` is 3, the clock enters the same painting loop it did before: painting first the time, then the day of the week, then the date. 

If ```VisibleLines``` is 1, it will also enter the same painting loop it did before: painting only the time.

![image](https://github.com/reactos/reactos/assets/16692240/210ef512-bbae-4ec3-ada3-71c203a8b638)

Reply:
What if `4`?

Reply:
4 is greater than ```CLOCKWND_FORMAT_COUNT``` so after it prints all three lines it exits the loop. When drawn, the clock is centered.

![image](https://github.com/reactos/reactos/assets/16692240/cefaf557-2788-46cd-a6e9-83d9782fa880)